staging/src/k8s.io/apiserver/pkg/server/filters/compression.go

```diff
@@ -58,6 +58,18 @@ func WithCompression(handler http.Handler, ctxMapper request.RequestContextMappe
 				return
 			}
 			compressionWriter.Header().Set("Content-Encoding", encoding)
+
+			// periodically flush writer
+			go func() {
+				tick := time.Tick(time.Second)
+				for impl := compressionWriter.(*compressionResponseWriter); !imp.compressorClosed(); {
```

Comment:
this isn't thread-safe, right? the read this triggers can race with the write that sets it to nil on close (same goes for the call to compressorClosed from `Flush()`)